Packet++/header/HttpLayer.h

```diff
@@ -709,7 +740,14 @@ namespace pcpp
 		 * this parameter isn't supplied or supplied as empty string (""), the default message for the status code will be set
 		 * @return True if setting the status code was completed successfully, false otherwise
 		 */
-		bool setStatusCode(HttpResponseStatusCode newStatusCode, std::string statusCodeString = "");
+		PCPP_DEPRECATED bool setStatusCode(const HttpResponseStatusCode& newStatusCode, const std::string& statusCodeString);
```

Comment:
ditto: let's keep the original method signature?

```cpp
PCPP_DEPRECATED bool setStatusCode(HttpResponseStatusCode newStatusCode, std::string statusCodeString = "");
```

And add a comment to the doxygen?

Reply:
done